reaper-server/src/main/scala/org/broadinstitute/reaper/server/edge/WorkflowTracker.scala

```diff
@@ -0,0 +1,35 @@
+package org.broadinstitute.reaper.server.edge
+
+import akka.NotUsed
+import akka.stream.scaladsl.Source
+import cromwell.api.model.WorkflowId
+
+import scala.concurrent.Future
+
+/**
+  * Component responsible for:
+  * <ol>
+  *   <li>Finding workflows in Cromwell ready for reaping
+  *   <li>Marking successfully-processed workflows as reaped to ensure they aren't pulled twice
+  *   <li>Marking workflows which fail to be reaped so that persistent problems can be detected
+  * </ol>
+  */
+trait WorkflowTracker {
+
+  /**
+    * Generate a stream of workflow IDs which are ready for reaping.
+    *
+    * Might be an infinite stream.
+    */
+  def pullWorkflowIds(): Source[WorkflowId, NotUsed]
+
```

Comment:
Should there also be a markWorkflowBeingReaped or Reaping?

Reply:
Maybe, depending on how its laid out... I'll add a TODO comment mentioning it